### PR TITLE
Add support for the Kotlin programming language.

### PR DESCRIPTION
**Pull request by dezgeg**
Kotlin is a programming language developed by JetBrains: http://kotlin.jetbrains.org/

### CODE REVIEW

**Comment by jolkdarr** [comment]
```.kts``` file extension shall be added:
- ```kt```, normal source files, 
- ```kts```, script files.

File: src/parsers/kotlin.rl [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    kotlin_identifier_entity ${ entity = KOTLIN_IDENTIFIER; } => kotlin_ecallback;
+    kotlin_keyword_entity    ${ entity = KOTLIN_KEYWORD;    } => kotlin_ecallback;
+    kotlin_operator_entity   ${ entity = KOTLIN_OPERATOR;   } => kotlin_ecallback;
+    ^(space | digit)         ${ entity = KOTLIN_ANY;        } => kotlin_ecallback;
+  *|;
+}%%
+
+/************************* Required for every parser *************************/
+
+/* Parses a string buffer with C/C++ code.
```
Comment (gkopff) [inline src/parsers/kotlin.rl]: Should this comment really refer to C/C++?

**Comment by jolkdarr** [comment]
`interface` keyword is missing.
`typealias` (since Kotlin 1.1) keyword is missing.

**Comment by alexey-tsvetkov** [comment]
Is there anything preventing this PR from merging?

**Pull request merged**